tendrl/ceph_integration/librbd_utils.py

```diff
@@ -0,0 +1,106 @@
+import multiprocessing
+import os
+import rados
+import rbd
+
+from gevent import Timeout
+from Queue import Empty
+
+RADOS_TIMEOUT = 20
+RADOS_NAME = 'client.admin'
+SRC_DIR = '/etc/ceph'
+
+
+class RbdOperationTimout(Exception):
+    pass
+
+
+class ClusterHandle():
+
+    def __init__(self, cluster_name):
+        self.cluster_name = cluster_name
+
+    def __enter__(self):
+        if SRC_DIR:
+            conf_file = os.path.join(SRC_DIR, self.cluster_name + ".conf")
+        else:
+            conf_file = ''
+
+        self.cluster_handle = rados.Rados(
+            name=RADOS_NAME,
+            clustername=self.cluster_name,
+            conffile=conf_file)
+        self.cluster_handle.connect(timeout=RADOS_TIMEOUT)
+
+        return self.cluster_handle
+
+    def __exit__(self, *args):
+        self.cluster_handle.shutdown()
+
+
+def rbd_operation(cluster_name, attributes):
+    err = None
+    p = None
+    queue = multiprocessing.Queue()
```

Comment:
@r0h4n  i am using multiprocessing for sub-process so i used multiprocessing queue